hatchet/readers/hpctoolkit_reader_latest.py

```diff
@@ -0,0 +1,421 @@
+                    name = name[1:]
+
+                name = name[: re.match("^MPI_[a-zA-Z_]+", name).end()]
+
+            self._functions[pFunction] = {
+                "id": pFunction,
+                "name": name,
+                "line": line,
+                "offset": offset,
+            }
+            if pFile:
+                self._functions[pFunction]["file_id"] = self._parse_source_file(
+                    meta_db, pFile
+                )["id"]
+            if pModule:
+                self._functions[pFunction]["module_id"] = self._parse_load_module(
+                    meta_db, pModule
+                )["id"]
+
+        return self._functions[pFunction]
+
+    def _parse_context(
+        self,
+        current_offset: int,
+        total_size: int,
+        parent: Node,
+        meta_db: bytes,
+        parent_time: int,
+    ) -> None:
+
+        final_offset = current_offset + total_size
+
+        while current_offset < final_offset:
+            (szChildren, pChildren, ctxId, _, lexicalType, nFlexWords) = safe_unpack(
+                "<QQLHBB", meta_db, current_offset
+            )
+
+            try:
+                my_time = self._summary_profile[ctxId][self._time_metric]
+            except:
+                my_time = None
+
+            if my_time is None:
+                current_offset += 32 + nFlexWords * 8  # TODO: can I move this up??
+                continue
+
+            if self._parent_percentage is not None and my_time / parent_time * 100.0 < self._parent_percentage:
+                current_offset += 32 + nFlexWords * 8  # TODO: can I move this up??
+                continue
+
+            if self._application_percentage is not None and my_time / self._total_execution_time * 100 < self._application_percentage:
+                current_offset += 32 + nFlexWords * 8  # TODO: can I move this up??
+                continue
+
+
+            frame = {
+                "id": ctxId,
+                "type": NODE_TYPE_MAPPING[lexicalType],
+                "depth": parent.frame["depth"] + 1,
+            }
+
+            if nFlexWords:
+                flex_offset = current_offset + 32
+
+                if lexicalType == 0:
+                    (pFunction,) = safe_unpack("<Q", meta_db, flex_offset)
+                    frame["name"] = self._parse_function(meta_db, pFunction)["name"]
+
+                elif lexicalType == 3:
+                    (pModule, offset) = safe_unpack("<QQ", meta_db, flex_offset)
+                    frame["name"] = (
+                        f"{self._parse_load_module(meta_db, pModule)['module_path']}:{offset}"
+                    )
+
+                else:
+                    (pFile, line) = safe_unpack("<QL", meta_db, flex_offset)
+                    frame["name"] = (
+                        f"{self._parse_source_file(meta_db, pFile)['file_path']}:{line}"
+                    )
+
+            node = Node(Frame(frame), parent=parent, hnid=ctxId)
```

Comment:
We usually do not set the `_hatchet_nid` field of `Node` objects using information from the profile. @pearce8 what should we use instead here?

Reply:
Addressed in Slack